### ISSUE TITLE
[Problem] Getting Wrong mAP Value on Raspberry Pi 4

### ISSUE THREAD

**Issue by jaskiratsingh2000** [jaskiratsingh2000/Research-Scalable-Vehicle-Detection-on-Edge-Devices#24]
I am trying to run the variant of the YOLO algorithm which is [Yolov5](https://github.com/ultralytics/yolov5) from [ultralytics](https://github.com/ultralytics) organization on GitHub. 
The Yolov5 project has an automatic script within the [val.py](https://github.com/ultralytics/yolov5/blob/master/val.py) file which let you see the metrics for the Accuracy Performance (mAP value) So, my problem is that when I am trying to run the same trained model on the Server, Ubuntu Machine, and Raspberry Pi 4, I am getting different value on Raspberry Pi 4 which should not happen. I am getting an mAP value of **0.564 on the Server** and **Ubuntu Machine** whereas I am getting **0.000127 on Raspberry Pi 4** which is almost negligible and the wrong value.

Here are the mAP value metrics for different machines:

**On Server:**

```
             Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95:
                 all       5000      36335      0.663      0.521      0.564      0.363
Speed: 0.4ms pre-process, 76.0ms inference, 8.4ms NMS per image at shape (32, 3, 640, 640)
```

**On Ubuntu Machine:**

```
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95:
                 all       5000      36335      0.668      0.518      0.564      0.363
Speed: 4.0ms pre-process, 1273.3ms inference, 21.3ms NMS per image at shape (32, 3, 640, 640)
```
**On Raspberry Pi 4:**

```
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95:
                 all       5000      36335    0.00163    0.00058   0.000127   6.94e-05
Speed: 6.2ms pre-process, 2178.0ms inference, 248.3ms NMS per image at shape (32, 3, 640, 640)
```

Below are the steps which tell about how I processed the mAP value.

* I trained the pre-trained model which is [yolov5s.yaml](https://github.com/ultralytics/yolov5/blob/master/models/yolov5s.yaml ) on the standard COCO dataset which has been given inside this Yolov5 repo as [coco.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml). If you are thinking that why I trined the already trained model, why I didn't use the already trained model for computing the mAP value. So I just want to let you know that for the very first time I used the "existing already trained" model only and it gave me the worst value even worsen than the above I got.
* FYI the training took place on some “X” server with the following command - 
```
python3 train.py --batch 50 --epochs 300 --data coco.yaml --weights yolov5s.pt --cfg yolov5.yaml 
```
which are the recommended parameters in their Github repo.
* After the trained model I received the “**best.pt**” and “**last.pt**” which are the weights file. These weights files I transferred into my Raspberry Pi 4.
* After these files were transferred into the Raspberry Pi 4, I ran the following command to test out the Accuracy Performance on the Raspberry Pi 4 - 
```
python3 val.py --data coco.yaml --weights best.pt
```
The val.py script has been made in such a way that it shows the metrics including the mAP value which can be seen above.

Hence the value that I receive is **0.000127** which is not correct. But if I run the same steps on the server I receive the **0.56** mAP value which is the correct

Any help would be appreciable. Thanks!

**Comment by glenn-jocher** [comment]
@jaskiratsingh2000 not sure what the reason might be. Can you outline steps to reproduce so users with RPi's might be able to assist?

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher I have already written the steps above with which I got the mAP value. I am really looking forward to the help. Do you think that I should try something else once? I am just wondering that why the "TFlite" version is getting more stability over Pytorch model in Raspberry Pi?
